miso-web/src/main/webapp/scripts/hot_sample.js

```diff
@@ -577,97 +577,100 @@ HotTarget.sample = (function() {
-            if (categories.length > 1) {
-              alert("You have selected samples of categories " + categories.join(" & ") + ". Please select samples from only one category.");
-              return;
-            }
+              if (samples.some(function(sample) {
+                return sample.sampleClassId;
+              }) && !Constants.isDetailedSample) {
+                alert("There's detailed samples, but MISO is not configured for this.");
+                return;
+              }
 
-            if (categories[0] == 'Tissue Processing' && classes.length > 1) {
-              alert("You have selected samples of classes " + classes.map(Utils.array.getAlias).join(" & ")
-                  + ". Please select samples from only one tissue processing class.");
-              return;
-            }
+              var classes = getSampleClasses(samples);
+              var categories = Utils.array.deduplicateString(classes.map(function(sampleClass) {
+                return sampleClass.sampleCategory;
+              }));
+              if (categories.length > 1) {
+                alert("You have selected samples of categories " + categories.join(" & ")
+                    + ". Please select samples from only one category.");
+                return;
+              }
 
-            window.location = "/miso/sample/bulk/edit?" + jQuery.param({
-              ids: samples.map(Utils.array.getId).join(',')
-            });
-          }
-
-        }, {
-          name: "Propagate",
-          action: function(samples) {
-            var idsString = samples.map(Utils.array.getId).join(",");
-            var classes = getSampleClasses(samples);
-
-            // In the case of plain samples, this will be empty, which is fine.
-            var targets = getChildSampleClasses(classes).sort(Utils.sorting.sampleClassComparator).map(function(sampleClass) {
-
-              return {
-                name: sampleClass.alias,
-                action: function(replicates) {
-                  window.location = "/miso/sample/bulk/propagate?" + jQuery.param({
-                    parentIds: idsString,
-                    replicates: replicates,
-                    sampleClassId: sampleClass.id
-                  });
-                }
-              };
-
-            });
-            if (!Constants.isDetailedSample || classes.every(function(sampleClass) {
-              return sampleClass.sampleCategory == "Aliquot";
-            })) {
-              targets.push({
-                name: "Library",
-                action: function(replicates) {
-                  window.location = "/miso/library/bulk/propagate?" + jQuery.param({
-                    ids: idsString,
-                    replicates: replicates
-                  });
-                }
+              if (categories[0] == 'Tissue Processing' && classes.length > 1) {
+                alert("You have selected samples of classes " + classes.map(Utils.array.getAlias).join(" & ")
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this usage of alert(...). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)